src/check.h

```diff
@@ -68,4 +69,11 @@ class CheckHandler {
 #define CHECK_GT(a, b) CHECK((a) > (b))
 #define CHECK_LT(a, b) CHECK((a) < (b))
 
+#define CHECK_EQ_EPS(a, b, eps) CHECK(std::fabs((a) - (b)) <  (eps))
```

Comment:
CHECK_FLOAT_EQ or CHECK_DOUBLE_EQ are more aligned with googletest.

alternatively, we might want to start using googletest ;) another TODO!